Code/Editor/CryEdit.cpp

```diff
@@ -1458,13 +1460,29 @@ void CCryEditApp::RunInitPythonScript(CEditCommandLineInfo& cmdInfo)
     using namespace AzToolsFramework;
     if (cmdInfo.m_bRunPythonScript || cmdInfo.m_bRunPythonTestScript)
     {
-        // cmdInfo data is only available on startup, copy it
-        QByteArray fileStr = cmdInfo.m_strFileName.toUtf8();
-
+        std::string fileStr;
         // We support specifying multiple files in the cmdline by separating them with ';'
+        // If a semicolon list of .py files is provided we look at the arg string
+        if (cmdInfo.m_strFileName.endsWith(".py"))
```

Comment:
This is a great solution..
Just having a different behaviour for cmdline argument based entirely on extension adds tight complexity for the future.

I would advise instead of doing this, to have a different cmdline argument reading the list from a filename